docs/resources/task.md

```diff
@@ -68,6 +68,13 @@ In addition to all arguments above, the following attributes are exported:
 - `events` - List of events for the machine orchestrator.
 - `logs` - List with task logs; one for each machine.
 
+After `terraform apply`, these attributes may be obtained using `terraform show` or `terraform console`. For example:
+
+```console
+$ terraform show -json | jq -r .values.root_module.resources[0].values.logs[0]
+$ echo "iterative_task.example.logs[0]" | terraform console
+```
+
```

Comment:
````suggestion
After `terraform apply`, these attributes may be obtained using `terraform console`:

```console
$ terraform console <<< 'join("\n", iterative_task.example.logs)'
```

Alternatively, add an `output` block to `main.tf` and use `terraform output --raw logs` to retrieve the logs:

```hcl
resource "iterative_task" "example" {
  ...
}

output "logs" {
  value = join("\n", iterative_task.example.logs)
}
```


````

Reply:
- any advantage of `<<<` over `echo |`?
- any reason to remove mention of `show -json` as an alternative?

Reply:
> _any advantage of `<<<` over `echo |`?_

Technically, rather the opposite: `echo |` is POSIX–compatible and `<<<` isn't. Visually, the latter is (arguably) clearer.

Reply:
> _any reason to remove mention of `show -json` as an alternative?_

Yes, the subsequent `jq` command was broken at every imaginable level. See https://github.com/iterative/terraform-provider-iterative/issues/389#issuecomment-1036090850 for the only correct way of parsing the output of `terraform show --json` with `jq`

Reply:
```console
terraform show --json | jq --raw-output '
  .values.root_module.resources[] |
  select(.address == "iterative_task.ca1sk") |
  .values.logs[]    
'
```

Reply:
The `jq` command you suggest only works as expected when there aren't any Terraform resources besides that `iterative_task` and it contains a single log, no more, no less.

Reply:
the "correct way" is probably still worth a mention. JSON is after all useful for building APIs on top of.

Reply:
> _the "correct way" is probably still worth a mention_

Sounds fair

> _JSON is after all useful for building APIs on top of_

Yes, let's build an API in top of Terraform ✈️

Reply:
````suggestion
After `terraform apply`, these attributes may be obtained using `terraform console`:

```console
$ terraform console <<< 'join("\n", iterative_task.example.logs)'
```

It's also possible to retrieve the raw output by using `terraform output --json` and `jq` like this:

```console
terraform show --json | jq --raw-output '
  .values.root_module.resources[] |
  select(.address == "iterative_task.example") |
  .values.logs[]
'
```

Alternatively, add an `output` block to `main.tf` and use `terraform output --raw logs` to retrieve the logs:

```hcl
resource "iterative_task" "example" {
  ...
}

output "logs" {
  value = join("\n", iterative_task.example.logs)
}
```
````

Reply:
@casperdcl, please take another look (and casperify the prose?)

Reply:
`apply` doth not worke with this

```
$ terraform apply
...
│ Error: Invalid function argument
│ 
│   on main.tf line 19, in output "logs":
│   19:   value = join("\n", iterative_task.casper_test.logs)
│     ├────────────────
│     │ iterative_task.casper_test.logs is null
│ 
│ Invalid value for "lists" parameter: argument must not be null.
```

Reply:
@casperdcl, edited, please try now. ♟️

Reply:
> _Note that such outputs get printed on apply and refresh (without needing show)._

Yes, unless you want the raw output, without any of the Terraform colors nor heredocs. In that case, you have to use `terraform output --raw`

Reply:
> _what 'bout `terraform show -json | jq -r '.values.root_module.resources[].values.logs | join("\n")'`?_

You probably mean `terraform show --json | jq --raw-output '.values.root_module.resources[].values.logs[]'`[^1]

```console
jq: error (at <stdin>:7): Cannot iterate over null (null)
```

[^1]: Functionally equivalent, but shorter.